### PR TITLE
Rynayae 207 cogs 402 page

### PR DESCRIPTION
**Pull request by ryanyae**
## Description
Made resources page, COGS 402 page, and added add resource tab.

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by ryanyae** [comment]
[resource pop up implementation (mobile view + tablet view + pc)](https://trello.com/c/x7WAntOX/209-resource-pop-up-implementation-mobile-view-tablet-view-pc)

**Comment by ryanyae** [comment]
[cogs 402 page (mobile view + tablet view + pc)](https://trello.com/c/LfEhLNAC/207-cogs-402-page-mobile-view-tablet-view-pc)

File: src/pages/Resources/Cards/ResourceCards.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    title: string;
+    link: string;
+}
+
+const ResourceCards = (props:PropsOBJ) => {
+
+    const {title, link} = props;
+    const history = useHistory();
+
+    console.log(link);
```
Comment (ElisaHu1) [inline src/pages/Resources/Cards/ResourceCards.tsx:14]: why do we need to log out the link here?

Comment (ryanyae) [reply src/pages/Resources/Cards/ResourceCards.tsx:14]: This is why we have code reviews! I completely missed that! Will change now!

Comment (ryanyae) [reply src/pages/Resources/Cards/ResourceCards.tsx]: Done!

File: src/pages/Resources/Page/ResourcePage.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            <AddResource isVisible={isAddMenuVisible} setVisible={setAddMenuVisible} />
+            <div style={{ filter: isAddMenuVisible ? 'blur(10px)' : 'blur(0px)', transition: 'all 0.5s ease-in-out', }}>
+                <div className='resource-page-container' style={{ marginTop: navHeight }}>
+                    <div className='resource-page-title-container'>
+                        <h1 className='resource-page-title'>
+                            {currentResource?.page_title}
+                        </h1>
+                    </div>
+                    {
+                        /* <div className='resource-content-containers' style={{ gridTemplateColumns: `repeat(${currentResource!.headers ? currentResource!.headers.length : 3}, 1fr)` }}>
```
Comment (ElisaHu1) [inline src/pages/Resources/Page/ResourcePage.tsx:60]: i see some of the code has been commented out, I wonder do you leave them here on purpose or we could delete them for clarity?

File: src/pages/Resources/Page/ResourcePage.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    const currentResource = RESOURCES.CONTENT.find((resource) => resource.title === match.params.resource_id);
+
+
+    const [navHeight, setNavHeight] = useState<number>(0);
+    const [filterDates, setFilterDates] = useState<string[]>([currDate.getFullYear().toString(), (currDate.getFullYear() - 1).toString(), (currDate.getFullYear() - 2).toString()]);
+    // const [currentFilter, setFilter] = useState<string>(currentResource!.headers ? currentResource!.headers[0] : filterDates[0]);
+    const [isAddMenuVisible, setAddMenuVisible] = useState<boolean>(false);
+
+    // useEffect(() => {
```
Comment (ElisaHu1) [inline src/pages/Resources/Page/ResourcePage.tsx:26]: same here

File: src/pages/Resources/Page/AddResource/AddResource.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        } else {
+            console.log('No file selected');
+        }
+    };
+
+    return (
+        <div className='add-resource-content-container' style={{ display: `${isVisible ? '' : 'none'}` }}>
+            <div className='add-resource-content'>
+                <div className='add-resource-close-tab-container'>
+                    {/* <div >
```
Comment (ElisaHu1) [inline src/pages/Resources/Page/AddResource/AddResource.tsx:28]: same here

File: src/pages/Resources/Page/ResourcePage.tsx (hunk 2 above)
Comment (ryanyae) [reply src/pages/Resources/Page/ResourcePage.tsx:60]: This should be taken out.

**Comment by ryanyae** [comment]
@ElisaHu1 The comments referring to the Resource.tsx file is all related. I just had a design question and was wondering if the filter bar is truly necessary?

**Comment by ElisaHu1** [comment]
> @ElisaHu1 The comments referring to the Resource.tsx file is all related. I just had a design question and was wondering if the filter bar is truly necessary?
> 
> image
> 
> The page is already sectioned for the different filters, is adding an extra filter bar really that necessary? Or should I make it so that It will auto-scroll to the components?

You are right! auto-scroll makes more sense!

**Review by ElisaHu1 — APPROVED** [review]
looks great! in later ticket we probably need to add checks for XML injection or illegal input detection on the backend

**Comment by ryanyae** [comment]
@ElisaHu1 Finished the autoscrolling behaviour!

**Comment by ElisaHu1** [comment]
> @ElisaHu1 Finished the autoscrolling behaviour! Do you happen to know where the content is that I can fill out? Just so that when I push this change it will have accurate information.

lets push the change and add content later! (also I do not know what to add lol)

**Pull request merged**